ci/prepare-unity.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+set -e -u -x -o pipefail
+
+cd "$(dirname "$0")/../"
+
+source ci/includes/pinned-tools.sh
+source ci/includes/profiling.sh
+
+function isLinux() {
+  [[ "$(uname -s)" == "Linux" ]];
+}
+
+function isMacOS() {
+  [[ "$(uname -s)" == "Darwin" ]];
+}
+
+function isWindows() {
+  ! ( isLinux || isMacOS );
+}
+
+# Only run inside TeamCity
+if [ -z "${TEAMCITY_CAPTURE_ENV+x}" ]; then
+    exit 0
+fi
+
+if isWindows; then
+    MODULES="linux mac-mono"
+elif isMacOS; then
+    MODULES="linux windows"
+elif isLinux; then
+    echo "Building is not supported on linux"
+    exit 1
+fi
+
+GOPATH="$(pwd)/go"
+export GOPATH
+UNITY_PACKAGE="github.com/improbable/unity_downloader"
+
+markStartOfBlock "$0"
+
+# Check if version is installed
+if isUnityImprobablePathPresent; then
+    exit 0
+fi
+
+# Setup unity_downloader
+markStartOfBlock "Installing unity_downloader"
+mkdir -p "${GOPATH}"
+go get -v -d -u "${UNITY_PACKAGE}"
```

Comment:
not too happy of introducing go into this repo as well...
also: this will fail for anyone who has not access to the unity downloader repo

Reply:
I think we need bigger signposts that, like `ci/create-windows-image.ps1`, this script only applies to internal Improbable infrastructure and shouldn't be run by the general public.